c2/c2-protocol/c2-protocol-api/src/main/java/org/apache/nifi/c2/protocol/api/ProcessorStatus.java

```diff
@@ -37,6 +37,7 @@ public class ProcessorStatus implements Serializable {
     private long processingNanos;
     private int activeThreadCount;
     private int terminatedThreadCount;
+    private String runState;
```

Comment:
Is there a reason for calling this `runState` as opposed to `runStatus` as named in the source Processor Status object?

Reply:
Not particularly, and I think we wanna go with runStatus instead. We did indeed implement it as runStatus in c++
https://github.com/apache/nifi-minifi-cpp/commit/07b9641ff033c1efb214d050079f4451df4e3eb9#diff-00076823b84623f3ab4d37455b5d6943cf6becc21da552be650affcae6344abdR39

To be honest, I only created this draft PR as a discussion starter, so we can see what would be the natural way to implement this in minifi java so we can implement the same thing in minifi c++.

But it turns out the C2 command to start stop components is only present in the C++ agents so this seems to be putting the cart before the horse in the case of the java agent. I will close this, and we can add this field when we implement the start/stop C2 command.

Reply:
Thanks for the background @martinzink, sounds good for now.